src/parquet/bloom_filter.cc

```diff
@@ -0,0 +1,168 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <algorithm>
+#include <cmath>
+#include <cstdint>
+
+#include "arrow/status.h"
+#include "arrow/util/bit-util.h"
+#include "parquet/bloom_filter.h"
+#include "parquet/bloom_filter_algorithm.h"
+#include "parquet/exception.h"
+#include "parquet/murmur3.h"
+#include "parquet/util/logging.h"
+
+namespace parquet {
+
+BloomFilter::BloomFilter(uint32_t num_bytes)
+    : hash_strategy_(HashStrategy::MURMUR3_X64_128), algorithm_(Algorithm::BLOCK) {
+  switch (hash_strategy_) {
+    case HashStrategy::MURMUR3_X64_128:
+      this->hasher_.reset(new MurmurHash3());
+      break;
+    default:
+      throw parquet::ParquetException("Unsupported hash strategy.");
```

Comment:
While you're in the `parquet` namespace, there is no need to prefix this with `parquet::`

Reply:
ok. Thanks.